EIPS/eip-custom_errors_for_erc_tokens.md

```diff
@@ -0,0 +1,401 @@
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+The provided interfaces with errors SHOULD conform with the [error grammar rules](#error-grammar-rules) described in the [rationale](#rationale) section.
+
+This EIP defines standard errors that may be used by implementations in certain scenarios, but does not specify whether implementations should revert in those scenarios, which remains up to the implementers, unless a revert is mandated by the corresponding EIPs.
+
+The names of the error arguments follow the [parameter glossary](#parameter-glossary) section, defining a clear guideline for assigning values to them.
+
+### [EIP-20](./eip-20.md)
+
+#### `ERC20InsufficientBalance(address sender, uint256 balance, uint256 needed)`
+
+Indicates an error related to the current `balance` of a `sender`.
+Used in transfers.
+
+- MUST be used when `balance` is less than `needed`.
+- MUST NOT be used if `balance` is equal or greater than `needed`.
+
+#### `ERC20InvalidSender(address sender)`
+
+Indicates a failure with the token `sender`.
+Used in transfers.
+
+- MUST be used for disallowed transfers from the zero address.
+- MUST NOT be used for approval operations.
+- MUST NOT be used for balance or allowance requirements.
+  - Use `ERC20InsufficientBalance` or `ERC20InsufficientAllowance` instead.
+
+#### `ERC20InvalidReceiver(address receiver)`
+
+Indicates a failure with the token `receiver`.
+Used in transfers
+
+- MUST be used for disallowed transfers to the zero address.
+- MUST be used for disallowed transfers to non-compatible addresses (eg. contract addresses).
+- MUST NOT be used for approval operations.
+
+#### `ERC20InsufficientAllowance(address spender, uint256 allowance, uint256 needed)`
+
+Indicates a failure with the `spender`'s `allowance`.
+Used in transfers.
+
+- MUST be used when `allowance` is less than `needed`.
+- MUST NOT be used if `allowance` is equal or greater than `needed`.
+
+#### `ERC20InvalidApprover(address approver)`
+
+Indicates a failure with the `approver` of a token to be approved.
+Used in approvals.
+
+- MUST be used for disallowed approvals from the zero address.
+- MUST NOT be used for transfer operations.
+
+#### `ERC20InvalidSpender(address spender)`
+
+Indicates a failure with the `spender` to be approved.
+Used in approvals.
+
+- MUST be used for disallowed approvals to the zero address.
+- MUST be used for disallowed approvals to the owner itself.
+- MUST NOT be used for transfer operations.
+  - Use `ERC20InsufficientAllowance` instead.
+
+### [EIP-721](./eip-721.md)
+
+#### `ERC721InvalidOwner(address sender, uint256 tokenId)`
```

Comment:
```suggestion
#### `ERC721InvalidOwner(address sender, address needed, uint256 tokenId)`
```

Reply:
Applied second suggestion with name `owner` on the last commit